### DISCUSSION TITLE
$GITHUB_PATH does not add to the path

### DISCUSSION THREAD

**Discussion by jameskr97** [community/community#26380] [Actions]
<p>Hi Github Community,</p>
<p>I’m having trouble using $GITHUB_PATH. I can’t tell if I’m not using it correctly, or if It’s not executing correctly.</p>
<p>For both of the following links, please pay attention to the Windows builds only.</p>
<p><a href="https://github.com/jameskr97/etterna/runs/1407721809" rel="noopener nofollow ugc">Here is a link</a> to a  previous version of a Github Action script that did work prior the switch to $GITHUB_PATH.</p>
<p><a href="https://github.com/jameskr97/etterna/runs/1414441535" rel="noopener nofollow ugc">Here is a link</a> to the most recent version of the Github Action script that stopped working with the $GITHUB_PATH changes.</p>
<p>Please note specifically the steps titled <code>Update Environment Variables</code> whose source is the following:<br>
This first code block is what worked successfully prior to the changed.</p>
<pre><code class="lang-yaml">- name: Update Environment Variables
  run: |
    echo "::add-path::${{github.workspace}}/main/extern/crashpad/breakpad/win/"
    echo "::set-env name=ETTERNA_ARCH::${{matrix.cfg.name}}"
</code></pre>
<p>The above code block was updated to below with the $GITHUB_PATH changes</p>
<pre><code class="lang-yaml">- name: Update Environment Variables
  run: |
    echo "${{github.workspace}}/main/extern/crashpad/breakpad/win/" &gt;&gt; $GITHUB_PATH
    echo "ETTERNA_ARCH=${{matrix.cfg.name}}" &gt;&gt; $GITHUB_ENV
</code></pre>
<p>Here is a following step titled <code>Generate Symbols</code> where that path is required:</p>
<pre><code class="lang-yaml">- name: Generate Symbols
  run: |
    echo "Dumping Symbols..."
    dump_syms.exe main/Program/Etterna.pdb &gt; ${{github.workspace}}/main/Etterna.sym
</code></pre>
<p><code>dump_syms.exe</code> is in the <code>${{github.workspace}}/main/extern/crashpad/breakpad/win/</code> folder.</p>
<p>Am I doing something incorrectly with the usage of the $GITHUB_PATH variable?</p>
<p>Thanks in advance,<br>
James</p>

**Comment by thboop** [comment, marked as answer]
<p>Hey <a class="mention" href="/jameskr97">@jameskr97</a>, you appear to be using the bash example, but on windows a step runs on pwsh (powershell core) by default.</p>
<p>Since you are using hosted runners, bash is included. Try updating your step to to run that step in bash:</p>
<pre><code class="lang-auto">- name: Update Environment Variables
  shell: bash
  run: |
    echo "${{github.workspace}}/main/extern/crashpad/breakpad/win/" &gt;&gt; $GITHUB_PATH
    echo "ETTERNA_ARCH=${{matrix.cfg.name}}" &gt;&gt; $GITHUB_ENV
</code></pre>
<p>If you need to be using powershell, you can follow the powershell example <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files" rel="noopener nofollow ugc">here</a></p>
<p>It would look like:</p>
<pre><code class="lang-auto">steps:
  - run: |
      echo "${{github.workspace}}/main/extern/crashpad/breakpad/win/" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
      echo "ETTERNA_ARCH=${{matrix.cfg.name}}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
</code></pre>

**Comment by jameskr97** [comment]
<p>Thank you very much!</p>
<p>I’m so tuned into bash, I didn’t think of the shell that I was using.</p>
<p>I appreciate your help.</p>

**Comment by RunningLeon** [comment]
@thboop Hi, sorry to bother you. On my self-hosted runner(windows10), the GITHUB_PATH variable cannot be effective in next steps, is there something wrong?
```
      - name: Setup
        run: |
          echo "$pwd\PYTHON_DIR" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
      - name: Test 
        run: |
          python -V # This line failed.
```

**Reply by jo-mue** [reply]
@RunningLeon I don't know if this applies to you but my first Append-Call to the GITHUB_PATH always prepends a '?'-char
My workaround includes first appending an empty string to GITHUB_PATH and then the actual path.
```
    - name: Setup
      run: |
        "" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
        "$pwd\PYTHON_DIR" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
      shell: powershell
    - name: echo path for validating (will be removed later)
      run: |
        echo $env:Path
      shell: powershell
      - name: Test 
        run: |
          python -V # This line failed.
```
Maybe someone has an explanation to this or an less workaroundish solution
